libs/configuration/src/lib/configuration/configuration.component.ts

```diff
@@ -1,11 +1,95 @@
-import { CommonModule } from '@angular/common';
-import { Component } from '@angular/core';
+import { AsyncPipe, NgIf } from '@angular/common';
+import { ChangeDetectionStrategy, Component, OnInit } from '@angular/core';
+import { FormBuilder, ReactiveFormsModule } from '@angular/forms';
+import { Project } from '@angular-cli-gui/shared/data';
+import { JsonObject } from '@angular-devkit/core/src/json/utils';
+import { FormlyFieldConfig, FormlyModule } from '@ngx-formly/core';
+import { FormlyJsonschema } from '@ngx-formly/core/json-schema';
+import { FormlyMaterialModule } from '@ngx-formly/material';
+import {
+  combineLatest,
+  forkJoin,
+  map,
+  Observable,
+  Subject,
+  switchMap,
+  tap,
+} from 'rxjs';
+
+import { WorkspaceSettingsService } from '../services';
 
 @Component({
   selector: 'cli-configuration',
   standalone: true,
-  imports: [CommonModule],
+  imports: [
+    NgIf,
+    AsyncPipe,
+    FormlyModule,
+    FormlyMaterialModule,
+    ReactiveFormsModule,
+  ],
   templateUrl: './configuration.component.html',
   styleUrls: ['./configuration.component.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class ConfigurationComponent {}
+export class ConfigurationComponent implements OnInit {
+  private formFields$ = new Subject<FormlyFieldConfig[]>();
+  private formData$ = new Subject<JsonObject>();
+  private projectName = '';
+
+  private readonly projectConfiguration$ = this.workspaceSettingsService
+    .readWorkspaceProjectConfiguration()
+    .pipe(
+      map(
+        (configuration) =>
+          this.schema.toFieldConfig(configuration)
+            .fieldGroup as FormlyFieldConfig[]
+      )
+    );
+
+  private readonly projectName$ = this.workspaceSettingsService
+    .readWorkspaceProjectNames()
+    .pipe(
+      map((names) => names[0]),
+      tap((projectName) => (this.projectName = projectName))
+    );
+
+  private readonly workspaceProject$ = this.projectName$.pipe(
+    switchMap((currentProjectName: string) =>
+      this.workspaceSettingsService.readWorkspaceProject(currentProjectName)
+    )
+  );
+
+  readonly formly$: Observable<{
+    formData: JsonObject;
+    formFields: FormlyFieldConfig[];
+  }> = combineLatest([this.formData$, this.formFields$]).pipe(
+    map(([formData, formFields]) => ({ formFields, formData }))
+  );
+
+  form = this.fb.group({});
+
+  constructor(
+    private workspaceSettingsService: WorkspaceSettingsService,
+    private fb: FormBuilder,
+    private schema: FormlyJsonschema
+  ) {}
+
+  ngOnInit(): void {
+    forkJoin([this.projectConfiguration$, this.workspaceProject$]).subscribe(
+      ([fields, projectData]) => {
+        this.formFields$.next(fields);
+        this.formData$.next(projectData);
+      }
+    );
+  }
+
+  onSubmit(): void {
+    this.workspaceSettingsService
+      .updateWorkspaceProjectConfiguration(
+        this.projectName,
+        this.form.value as Project
+      )
+      .subscribe();
```

Comment:
```suggestion
    this.projectName$.pipe(
      take(1),
      switchMap(projectName => 
        this.workspaceSettingsService
        .updateWorkspaceProjectConfiguration(
          projectName,
          this.form.value as Project
        )
      )
    ).subscribe();
```
then you can remove `this.projectName` and the tap that assigns its value